File: python/copass-management/src/copass_management/tools/list_sources.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def list_sources(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    provider = input.get("provider")
    return await ctx.client.sources.list(
        ctx.sandbox_id,
        provider=provider if isinstance(provider, str) else None,
    )

File: typescript/packages/management/src/tools/list_sources.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const listSources: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  return ctx.client.sources.list(ctx.sandboxId, {
    provider: typeof input.provider === 'string' ? input.provider : undefined,
  });
};
